### DISCUSSION TITLE
Compared to vitest snapshot testing of json swift-snapshot-testing is slow

### DISCUSSION THREAD

**Discussion by doozMen** [pointfreeco/swift-snapshot-testing#756] [General]
I realise this is not a very good description I it is more a question if you have an idea why a simple json rendering test is slow compared to vtest doing the same thing?

Any idea how to speed up snapshot tests compared to a test with a hardcoded string value?

For example in my tests a test like below tripled in duration when converted to snapshots

```swift
func testEncode() {
  let sut = try JSONEncoder().encode(Foo.self)

  let expected = """
  {
  ...
  }
  """
  XCAssertEqual(sut, expected)
````

**Comment by mbrandonw** [comment]
Hi @doozMen, since this is more of a question than an issue I am going to convert it over to a GitHub discussion.

**Comment by doozMen** [comment]
It seams like my problem related mostly to file access. The InlineSnapshotTesting is much faster and as I mostly use it to test generated code, so text, this solves my debate. I will close this one then.
